exmg-markdown-editor.ts

```diff
@@ -0,0 +1,927 @@
+    clearTimeout(timer);
+    if (cb) {
+      timer = window.setTimeout(cb, time);
+    }
+  };
+};
+
+/**
+* Markdown WYSIWYG editor element.
+* This editor element is a wrapper element for the markdown-element which will enable editing
+* the markdown data. See [marked-element](https://www.webcomponents.org/element/PolymerElements/marked-element/)
+* for more details on how to use this element.
+*
+* ```
+* <exmg-markdown-editor markdown="{{markdown}}">
+*   <marked-element markdown="{{markdown}}">
+*     <div slot="markdown-html"></div>
+*     <script type="text/markdown"\>
+*     # Header
+*     ...
+*     </script\>
+*   </marked-element>
+* </exmg-markdown-editor>
+* ```
+*
+* ## Custom Toolbar
+* Add attribute toolbar-buttons to adjust the toolbar buttons. The array values should match
+* the _toolbarButtons item name values.
+*
+* ```html
+* <exmg-markdown-editor toolbar-buttons='["strong","italic","strikethrough","|","quote","hr","table"]'>
+*  ...
+* </exmg-markdown-editor>
+* ```
+*
+* ### Styling
+*
+* The preview panel markdown output can be styled from outside th element. See the demo/styling.html example
+* on how to do this. In this demo the github-markdown-css project is used for styling the html output.
+*
+* `<exmg-markdown-editor>` provides the following custom properties and mixins
+*  for styling:
+*
+*  Custom property | Description | Default
+*  ----------------|-------------|----------
+*  `--exmg-markdown-editor` | editor mixin | `{}`
+*  `--exmg-markdown-editor-border` | Border Color | `#ddd`
+*  `--exmg-markdown-editor-background-color` | Editor Background Color | `white`
+*  `--exmg-markdown-editor-fullscreen-top-offset` | Top offset in fullscreen mode | `0px`
+*  `--exmg-markdown-editor-toolbar` | editor mixin | `{}`
+*  `--exmg-markdown-editor-toolbar-background` | Toolbar background color | `#fafafa`
+*  `--exmg-markdown-editor-toolbar-color` | Toolbar text color | `87% black`
+*  `--exmg-markdown-editor-toolbar-color-disabled` | Toolbar text color disabled | `54% black`
+*  `--exmg-markdown-editor-preview` | HTML Preview mixin | `{}`
+*  `--exmg-markdown-editor-preview-background` | Preview background color | `white`
+*  `--exmg-markdown-editor-toolbar-button-icon` | Toolbar button icon mixin | `{}`
+*  `--exmg-markdown-editor-toolbar-button-hover` | Toolbar button mixin | `{}`
+*  `--exmg-markdown-editor-toolbar-button-background-hover` | Toolbar icon border color | `#fafafa`
+*  `--exmg-markdown-editor-toolbar-seperator-color` | Toolbar seperator color | `#ddd`
+*  `--exmg-markdown-editor-code` | CodeMirror root mixin | `{}`
+*  `--exmg-markdown-editor-code-hover` | Editor code part hover background color | `white`
+*  `--exmg-markdown-editor-code-focused` | CodeMirror editor focused mixin | `{}`
+*
+*  # Events:
+*  - value-change - where detail is current markdown value
+*  - exmg-markdown-editor-fullscreen where detail is boolean with current fullscreen state
+*
+* @customElement
```

Comment:
I think these annotations are for polymer element, not for lit element

Reply:
More of them are just JsDoc so can be used regardless of framework.
I will keep it as it was